.gitignore

```diff
@@ -11,3 +11,5 @@ documentation
 .metadata
 .pnp
 .pnp.js
+.settings/org.eclipse.core.resources.prefs
+.settings/org.eclipse.m2e.core.prefs
```

Comment:
You can put whole `.settings` in `.gitignore` I'm sure other files will pop up in it.